classes/author/Collector.inc.php

```diff
@@ -1,54 +1,99 @@
 <?php
 /**
- * @file classes/services/QueryBuilders/PKPAuthorQueryBuilder.php
+ * @file classes/author/Collector.inc.php
  *
  * Copyright (c) 2014-2021 Simon Fraser University
  * Copyright (c) 2000-2021 John Willinsky
  * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
  *
- * @class PKPAuthorQueryBuilder
- * @ingroup query_builders
+ * @class author
  *
- * @brief Class for building database queries for authors
+ * @brief A helper class to configure a Query Builder to get a collection of announcements
  */
 
-namespace PKP\services\queryBuilders;
+namespace PKP\author;
 
+use Illuminate\Database\Query\Builder;
 use Illuminate\Support\Facades\DB;
-
+use PKP\core\interfaces\CollectorInterface;
 use PKP\plugins\HookRegistry;
-use PKP\services\queryBuilders\interfaces\EntityQueryBuilderInterface;
 
-class PKPAuthorQueryBuilder implements EntityQueryBuilderInterface
+class Collector implements CollectorInterface
 {
-    /** @var array get authors for one or more contexts */
-    protected $contextIds = [];
+    public const ORDERBY_SEQUENCE = 'a.seq';
+    public const ORDERBY_ID = 'a.author_id';
+
+    /** @var string The default orderBy value for authors collector */
+    public $orderBy = self::ORDERBY_SEQUENCE;
+
+    /** @var DAO */
+    public $dao;
+
+    /** @var array|null */
+    public $contextIds = null;
+
+    /** @var array|null */
+    public $publicationIds = null;
 
     /** @var string get authors with a family name */
     protected $familyName = '';
 
     /** @var string get authors with a given name */
     protected $givenName = '';
 
-    /** @var array get authors for one or more publications */
-    protected $publicationIds = [];
-
     /** @var string get authors with a specified country code */
     protected $country = '';
 
     /** @var string get authors with a specified affiliation */
     protected $affiliation = '';
 
+    /** @var int */
+    public $count = 0;
+
+    /** @var int */
+    public $offset = 0;
+
+    /** @var bool|null */
+    public $includeInBrowse = null;
+
+    public function __construct(DAO $dao)
+    {
+        $this->dao = $dao;
+    }
+
     /**
-     * Filter by one or more contexts
-     *
-     * @param array|int $contextIds
-     *
-     * @return \PKP\services\queryBuilders\PKPAuthorQueryBuilder
+     * Filter by contexts
+     */
+    public function filterByContextIds(array $contextIds): self
```

Comment:
Should be `?array`, indicating that the property can also be reset to `null` (indicating that the condition should not apply). This is distinguished from an empty array, which should indicate that no values are acceptable and would result in an empty result set.

Reply:
(Also true for other setters)

Reply:
Changed